### DISCUSSION TITLE
How to hide selected lines of code in quarto code chunk ?

### DISCUSSION THREAD

**Discussion by Andrzej-Andrzej** [quarto-dev/quarto-cli#5445] [Q&A]
Hi,
I want to  hide selected lines of code (not always consecutive) in quarto code chunk but still evaluate it and echo it in the output.
How can I do it ?

**Comment by baptiste** [comment]
````
---
title: "Untitled"
---

```{r, echo=c(2,3)}
1+1
2+2
3+3
4+4
```

````

**Reply by baptiste** [reply]
(or `echo=-c(2,3)` to selectively hide)

**Comment by Andrzej-Andrzej** [comment]
Thank you, should I take the line numbers from source editor or does every chunk have line numbering on its own ?

**Reply by baptiste** [reply]
The line numbers belong to the chunk

**Comment by Andrzej-Andrzej** [comment]
Thank you, I have tried it but it does not work. 
```
{r, echo = -c(5, 10, 15, 20}
```
I want to hide this line, but preserve the effect of this fucntion:
```
officer::run_linebreak()
```

**Reply by mcanouil** [reply]
Could you share a small working reproducible example and possibly a screenshot of what output you expect?

**Comment by seeM** [comment]
Is it possible to do this in Python / with the `jupyter` engine?

**Reply by seeM** [reply]
For more context, I want to use the `%%bash` cell magic in my notebook like so:

```sh
%%bash
run-some \
    multiline-command
```

But I want it to only display the command without the `%%bash` like:

```sh
run-some \
    multiline-command
```

**Reply by cderv** [reply]
We don't have this option. It is quite specific to **knitr**, and supported by Quarto through it. You would need to hack through Lua for supporting this I guess. 

For example : 

````lua
CodeBlock = function(el)
  if el.classes:includes("cell-code") then
    el.text = string.gsub(el.text, "%%%%%a+\n", "")
  end
  return el
end
````

used with 
````yaml
filters:
  - remove-cell-magic.lua
````

It could more scoped to be applied on a class or attributes put on the cell code. and Regex could probably be improved. 

Hope it helps

**Comment by dragonstyle** [comment]
This might also be helpful:

https://github.com/jjallaire/code-visibility

**Reply by cderv** [reply]
Oh nice extension !

**Reply by seeM** [reply]
Awesome, thank you both! 😄
